graylog2-web-interface/src/components/configurations/TimeRangeOptionsForm.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import { Row, Col, Button } from 'react-bootstrap';
+import ISODurationUtils from 'components/common/ISODurationUtils';
+
+import ObjectUtils from 'util/ObjectUtils';
+
+const TimeRangeOptionsForm = React.createClass({
+  propTypes: {
+    options: React.PropTypes.object,
+    title: React.PropTypes.string.isRequired,
+    help: React.PropTypes.any.isRequired,
+    addButtonTitle: React.PropTypes.string,
+    update: React.PropTypes.func.isRequired,
+    validator: React.PropTypes.func,
+  },
+
+  mixins: [ISODurationUtils],
+
+  getDefaultProps() {
+    return {
+      options: {},
+      addButtonTitle: 'Add option',
+      validator: () => true,
+    };
+  },
+
+  getInitialState() {
+    return {optionsList: this._buildOptionsState(this.props.options)};
+  },
+
+  componentWillReceiveProps(newProps) {
+    this.setState({optionsList: this._buildOptionsState(newProps.options)});
+  },
+
+  // Converts the object to a list of objects to make form handling easier.
+  _buildOptionsState(options) {
+    let optionsList = null;
+
+    if (options) {
+      optionsList = Object.keys(options).map((key) => {
+        return {period: key, description: options[key]};
+      });
+    }
+
+    return optionsList;
+  },
+
+  _update(options) {
+    const update = {};
+
+    options.forEach((option) => {
+      update[option.period] = option.description;
+    });
+
+    this.props.update(update);
+  },
+
+  _onAdd() {
+    const options = ObjectUtils.clone(this.state.optionsList);
+
+    if (options) {
+      options.push({period: 'PT0S', description: ''});
```

Comment:
On the second commit, the default period changed from 'PT1S' to 'PT0S', and that makes two things go wrong:

1. Adding an option to the relative range options ends in a no-result, as the `PT0S` option is already there
2. Adding an option to the time range options makes the default value to be invalid (and displaying the field in red)

I think we shouldn't use the period as key in this case, because it makes hard to pick default values, and also merges options that use the same period as another one had. Even if it makes no sense having two options with the same period, they should still have their own field where we should indicate this situation.